src/frontend/state/InjectedProviderHandler.ts

```diff
@@ -0,0 +1,159 @@
+import { makeAutoObservable } from 'mobx'
+
+class InjectedProviderHandler {
+  ethereumListenersBound = false
+
+  async handleRequest(
+    event: Electron.IpcRendererEvent,
+    id: number,
+    args: Record<string, unknown>
+  ) {
+    try {
+      await this.waitForConnection()
+      /* eslint-disable-next-line */
+      //@ts-ignore
```

Comment:
Nitty-picky but ts-ignore should have a reason after it, then you won't need the eslint-disable-next-line above it.